### ISSUE TITLE
Using numpy.squeeze() changes Dataset shape

### ISSUE THREAD

**Issue by valp124** [skjerns/mat7.3#55]
**Describe the bug**
I have a database of shape:

![image](https://github.com/user-attachments/assets/b4ee5764-e3b2-41df-a132-355c33414298)

and it's important that it is kept that way. Your code does `array.T.squeeze()` when reading a dataset, and I lose the axis of length 1, which is a big problem in my case. I'm not sure why you're doing the squeeze in the first place, but perhaps if that's necessary in certain cases, at least a boolean flag can be provided at the top level to control that.

Thank you!

**Comment by valp124** [comment]
Thanks a lot for the quick reply. Would you like me to make a PR introducing a top-level argument, something like `squeezeSingularAxis`?

**Comment by skjerns** [comment]
Best would be if data is by default represented as it is in MATLAB. The problem is here, as far as I remember, that it is sometimes difficult to differentiate between a (X) matrix and a (X, 1) matrix. But I think that should be possible to figure out.

I do have some test cases for this in the library. You can simply remove the `squeeze` and see if you can figure out how to remove dimensions that are not visible in MATLAB.

```
E           AssertionError:
E           Arrays are not equal
E
E           (shapes (5, 1, 1), (5,) mismatch)
E            x: array([[[1.]],
E
E                  [[2.]],...
E            y: array([1, 2, 3, 4, 5])

..\..\anaconda3\Lib\contextlib.py:81: AssertionError
```

**Comment by valp124** [comment]
So we could do something like this, I'm not sure if that's robust enough. 

```
def selective_squeeze(arr):

# Assumption: "superfluous" singular dimensions come at the end and are superfluous only if after removing them, we get a shape of 1

    shape = arr.shape
    
    if len(shape) <= 1:
        return arr

    # Iterate from the end of the shape and remove trailing dimensions of size 1
    while shape and shape[-1] == 1:
        shape = shape[:-1]

    # If removing all singular dimensions leaves a scalar or more than 1D array, return the original
    if len(new_shape) != 1:
        return arr

    return arr.reshape(new_shape)
```

**Comment by valp124** [comment]
Ok, I'm ready to make a PR but am not allowed to push to the repo. I have the fix, along with the test code and test .mat file.

**Comment by valp124** [comment]
> Ok, I'm ready to make a PR but am not allowed to push to the repo. I have the fix, along with the test code and test .mat file.

Ok, I didn't know I could PR from a fork.

**Comment by skjerns** [comment]
I had a look at it again and it seems like the problem is that MATLAB represents vectors as 2D-matrices. 

Therefore it should be enough to only strip the singular dimension if and only if the matrix has two dimensions and one of them is 1. However, there's not much we can do in terms of finding out if it's a column vector or a row vector, as in numpy those are the same. If dimensions>2 all dimensions should be loaded. I'm writing some test cases for that and implement another fix.

**Comment by valp124** [comment]
Yeah, that thought also occurred to me while I was debugging. Thank you so much for looking further into this!

**Comment by skjerns** [comment]
fixed by #60 :) now it should be consistent!

thanks for bringing this up! I'm surprised nobody else encountered this problem yet.
